todo_app/network/trello_client.py

```diff
@@ -1,60 +1,62 @@
-import os
 import requests
 
-BOARD_ID = os.getenv('TRELLO_BOARD_ID')
-
-LIST_IDS = {
-    'To do': None,
-    'Done': None
-}
-
-def url_builder(route):
-    return f"{os.getenv('TRELLO_API_BASE_URL')}{route}?key={os.getenv('TRELLO_API_KEY')}&token={os.getenv('TRELLO_API_TOKEN')}"
-
-def get_list_ids():    
-    url = url_builder(f"/boards/{BOARD_ID}/lists")
-    r = requests.get(url)
-    
-    LIST_IDS['To do'] = next(l['id'] for l in r.json() if l['name'] == 'To do')
-    LIST_IDS['Done'] = next(l['id'] for l in r.json() if l['name'] == 'Done')
-
-get_list_ids()
-
-def get_trello_lists_with_cards():
-    payload = {
-        'cards': 'open'
-    }
-    url = url_builder(f"/boards/{BOARD_ID}/lists")
-    r = requests.get(url, params=payload)
-    return r.json()
-
-def add_item_to_list(title, description):
-    url = url_builder(f"/cards")
-    payload = {
-        'name': title,
-        'desc': description,
-        'idList': LIST_IDS['To do']
-    }
-    r = requests.post(url, params=payload)
-    return r.json()
-
-def complete_card(id):
-    url = url_builder(f"/cards/{id}")
-    payload = {
-        'idList': LIST_IDS['Done']
-    }
-    r = requests.put(url, params=payload)
-    return r.json()
-
-def uncomplete_card(id):
-    url = url_builder(f"/cards/{id}")
-    payload = {
-        'idList': LIST_IDS['To do']
-    }
-    r = requests.put(url, params=payload)
-    return r.json()
-
-def delete_card(id):
-    url = url_builder(f"/cards/{id}")
-    r = requests.delete(url)
-    return r.json()
\ No newline at end of file
+class TrelloClient:
+    def __init__(self, board_id, trello_api_base_url, trello_api_key, trello_api_token):
+        self.BOARD_ID = board_id
+        self.TRELLO_API_BASE_URL = trello_api_base_url
+        self.TRELLO_API_KEY = trello_api_key
+        self.TRELLO_API_TOKEN = trello_api_token
+        self.LIST_IDS = {
+            'To do': None,
+            'Done': None
+        }
+        self.get_list_ids()
```

Comment:
After a bit of digging, I think this is one of the reasons why the integration test is failing. You're making a trello request before the stubbing logic is called, so in here you're not mocking the request, you're actually doing a request to trello.

I think the easiest way of solving this would be to remove it from the constructor, and call it in the methods below, when necessary, in the form of e.g.:
```
if self.LIST_IDS['To Do'] is None or self.LIST_IDS['Done'] is None:
    self.get_list_ids()
```

Reply:
Whoops! Thank you, difficult to spot an issue like this when you have the API keys set up locally so the request works